### PR TITLE
Feature/videos added unit tests

### PR DESCRIPTION
**Pull request by radutipurita**
## Summary
This PR adds unit tests for both the VideoViewModel and VideoRepositoryStorage classes to improve coverage and ensure correct functionality of video upload and retrieval operations.

#### VideoViewModelTest:

- uploadVideo should update uploadSuccess on success:
Verifies that uploadSuccess is correctly updated when a video upload succeeds.
- uploadVideo should update error on failure:
Ensures that error is updated when a video upload fails.
- loadVideos should update videoUrls on success:
Tests that the list of video URLs is correctly populated when video URLs are successfully loaded.
- loadVideos should update error on failure:
Ensures that the error field is updated when there is a failure in loading videos.
#### VideoRepositoryStorageTest:

- getVideoUrls should return all video URLs:
Ensures that getVideoUrls returns a list of all video URLs stored in Firebase and does not trigger the failure callback in a successful scenario.

#### Small side note:
Documentation was added as suggested.

### CODE REVIEW

File: app/src/test/java/com/android/sample/model/video/VideoRepositoryTest.kt [hunk 1]
```diff
@@ -1 +1,47 @@
...
+  private lateinit var mockListResult: ListResult
+
+  @Before
+  fun setup() {
+    mockStorageReference = Mockito.mock(StorageReference::class.java)
+    mockListResult = Mockito.mock(ListResult::class.java)
+    videoRepository = VideoRepositoryStorage(mockStorageReference)
+  }
+
+  @Test
```
Comment (Shayan105) [inline app/src/test/java/com/android/sample/model/video/VideoRepositoryTest.kt:25]: Test well structured and clear

**Review by Siupahc — COMMENTED** [review]
The way you do your tests are really good and well structured, and the comments are clear. Really good

**Review by Shayan105 — APPROVED** [review]
Comments are clear and concise. Test are well structured.  Approved !

**Comment by Shayan105** [comment]
I've just understood how the PR comment work. (I have to comment bellow the interesting region, not above) Sorry if the snippet showd by github is strange.

**Comment by JohnTomasTaylor** [comment]
As your teamates have already noted, this PR is really well written. 
The PR description is very clear, it has a clear purpose, the tests are clean and the code is well documented.
If these tests came bundled with #18 it would be a perfect one in my book. Good job!

File: app/src/test/java/com/android/sample/model/video/VideoRepositoryTest.kt [hunk 2]
```diff
@@ -1 +1,47 @@
...
+    val mockUriTask1: Task<Uri> = Tasks.forResult(Uri.parse("https://mock-video-url1.com"))
+    val mockUriTask2: Task<Uri> = Tasks.forResult(Uri.parse("https://mock-video-url2.com"))
+    val mockVideoRef1: StorageReference = Mockito.mock(StorageReference::class.java)
+    val mockVideoRef2: StorageReference = Mockito.mock(StorageReference::class.java)
+    Mockito.`when`(mockStorageReference.child("template_videos")).thenReturn(mockStorageReference)
+    Mockito.`when`(mockStorageReference.listAll()).thenReturn(Tasks.forResult(mockListResult))
+    Mockito.`when`(mockListResult.items).thenReturn(listOf(mockVideoRef1, mockVideoRef2))
+    Mockito.`when`(mockVideoRef1.downloadUrl).thenReturn(mockUriTask1)
+    Mockito.`when`(mockVideoRef2.downloadUrl).thenReturn(mockUriTask2)
+
```
Comment (JohnTomasTaylor) [inline app/src/test/java/com/android/sample/model/video/VideoRepositoryTest.kt:37]: I'm having a small doubt that this properly tests getVideoUrls, is it sure that the onSuccessListeners are properly called with an already completed task? I tried it on my side with an assert(false) in the success callback and the test passed.
If you have the time you could check to be sure, however don't lose to much time on this.

File: app/src/test/java/com/android/sample/model/video/VideoRepositoryTest.kt [hunk 3]
```diff
@@ -1 +1,47 @@
...
+    videoRepository = VideoRepositoryStorage(mockStorageReference)
+  }
+
+  @Test
+  fun getVideoUrls_shouldReturnAllVideoUrls() {
+    // Arrange
+    val mockUriTask1: Task<Uri> = Tasks.forResult(Uri.parse("https://mock-video-url1.com"))
+    val mockUriTask2: Task<Uri> = Tasks.forResult(Uri.parse("https://mock-video-url2.com"))
+    val mockVideoRef1: StorageReference = Mockito.mock(StorageReference::class.java)
+    val mockVideoRef2: StorageReference = Mockito.mock(StorageReference::class.java)
```
Comment (JohnTomasTaylor) [inline app/src/test/java/com/android/sample/model/video/VideoRepositoryTest.kt:31]: I'm having a small doubt that this properly tests getVideoUrls, is it sure that the onSuccessListeners are properly called with an already completed task? I tried it on my side with an assert(false) in the success callback and the test passed.
If you have the time you could check to be sure, however don't lose to much time on this.

**Pull request merged**